games/xworld3d/xworld3d_simulator.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2017 Baidu Inc. All Rights Reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <deque>
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <memory>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include "simulator.h"
+#include "xworld3d_flags.h"
+
+namespace simulator {
+namespace xworld3d {
+
+class X3SimulatorImpl;
+
+class X3Simulator : public GameSimulator {
```

Comment:
Why not inherit from MultiGameSimulator?

Reply:
i thought we were going to abandon MultiGameSimulator